.github/workflows/sanitisers.yml

```diff
@@ -0,0 +1,200 @@
+name: Sanitiser Checks
+
+# on: workflow_dispatch
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  conan-cache:
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+    env:
+      HOST_TYPE: ci
+      REDIS_QUEUE_HOST: redis
+      REDIS_STATE_HOST: redis
+    container:
+      image: faasm/faabric:0.2.2
+    defaults:
+      run:
+        working-directory: /code/faabric
+    services:
+      redis:
+        image: redis
+    steps:
+      # --- Code update ---
+      - name: "Fetch ref"
+        run: git fetch origin ${GITHUB_REF}:ci-branch
+      - name: "Check out branch"
+        run: git checkout --force ci-branch
+      # --- Set-up ---
+      - name: "Ping redis"
+        run: redis-cli -h redis ping
+      # --- Build dependencies
+      - name: "Build dependencies to be shared by all sanitiser runs"
+        run: inv dev.cmake -b Debug
+      - name: "Upload built CMake dependencies"
+        uses: actions/upload-artifact@v2
+        with:
+          name: cmake-cache
+          path: /build/faabric
```

Comment:
The conan cache is actually in `/root/.conan`, and it is hashed with build configuration so should generally work well, while caching the cmake output could cause problems (and you force a clean run which deletes the contents of this folder anyway)

Reply:
I disabled the clean build (afaict) but it was still not working. I decided to look at it in another moment, but thanks for pointing this out :grinning:

Reply:
Sorry for the wrong information, I just checked the logs and apparently on the actions runners the folder is at ` /github/home/.conan/`, I guess `$HOME/.conan` could work if standard environment variables can work properly in that context? It is `/root/.conan` in the docker build context though.

Reply:
You can also use `conan config home` to print out the path to shell reliably

Reply:
Tilde substitution [should work](https://github.com/actions/upload-artifact#environment-variables-and-tilde-expansion) for the `path` variable of the `upload-artifact` action. Let's see.

Reply:
It looks like the caching system doesn't handle this many output files very well, it's been uploading for a few minutes now and it's only at 8%. Tarring the folder like in <https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files> and untarring it on extract could help? If it doesn't, the upload takes longer than rebuilding the dependencies so I'm not sure if it's worth it

Reply:
@csegarragonz I'm happy to tinker around with this and try to fix it if you're working on other things, I'm somewhat interested in running the sanitizers myself too

Reply:
@kubasz i'm working on other things, and every now and then checking this; so very glad if you can take over :smile: